stack_overflow/src/main/java/com/gabrielcireap/stackOverflow/service/QuestionManagementService.java

```diff
@@ -0,0 +1,120 @@
+package com.gabrielcireap.stackOverflow.service;
+
+import com.gabrielcireap.stackOverflow.entity.Answer;
+import com.gabrielcireap.stackOverflow.entity.Question;
+import com.gabrielcireap.stackOverflow.entity.Tag;
+import com.gabrielcireap.stackOverflow.exception.NotEnoughPermissionsException;
+import com.gabrielcireap.stackOverflow.exception.QuestionNotFoundException;
+import com.gabrielcireap.stackOverflow.exception.TagNotFoundException;
+import com.gabrielcireap.stackOverflow.repository.RepositoryFactory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.sql.Timestamp;
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+public class QuestionManagementService {
+
+    private final RepositoryFactory repositoryFactory;
+    private final UserManagementService userManagementService;
+
+    @Transactional
+    public List<Question> listQuestions(){
+        userManagementService.checkIfUserIsLogged();
+        List<Question> questions = repositoryFactory.createQuestionRepository().findAll();
+        questions.sort((q1,q2) ->  q1.getCreationDate().after(q2.getCreationDate()) ? -1 : 1);
+        return questions;
+    }
+
+    @Transactional
+    public Map<Question, List<Answer>> findById(int id){
```

Comment:
You had to resort to this structure (which is more like a Pair than a Map) because your Question entity does not have the link towards the Answers :)